fix: fix missing swagger definitions [DET-4213]

hamidzr opened the pull request:
## Description

- avoid renaming path parameters that was causing a collision
- fix a separate make dependency issue with swagger spec generation


<!---
Lead with the intended commit body in this description field. For breaking changes,
please include "BREAKING CHANGE:" at the beginning of your commit body.
At minimum, this section should include a bracketed reference to the Jira ticket,
e.g. "[DET-1234]". When squash-and-merging, copy this directly into the description field.
-->


## Test Plan

- generated and compared api.swagger.json with and without the changes here and looked at the diff.

<!---
Describe the situations in which you've tested your change, and/or a screenshot as appropriate.
Reviewers may ask questions of this test plan to ensure adequate manual coverage of changes.
-->


## Commentary (optional)

<!---
Use this section of your description to add context to the PR. Could be for particularly
tricky bits of code that could use extra scrutiny, historical context useful for reviewers, etc.
You may intentionally leave this section blank and remove the title.
--->


## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.


<!---
## Title

Example title: "docs: tweak recommended "pip install" usage".

Specifically, this title should contain a type and a description
of the change being made:

User-facing change types:
- docs: docs-only change
- feat: new user-facing feature
- fix: bug fix
- perf: performance improvement

Internal change types:
- build: build system change (anything in a `Makefile`, mostly)
- chore: any internal change not covered by another type
- ci: anything that touches `.circleci`
- refactor: internal refactor
- style: style change
- test: new tests

See https://www.conventionalcommits.org/en/v1.0.0/ for background.

The first line should also:
- be at most 89 characters long
- contain a description that is at most 72 characters long
- not end with sentence-ending punctuation
- start (after the type) with a lowercase imperative ("add", "fix")
-->

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

stoksc commented on proto/Makefile:
```diff
@@ -42,8 +42,8 @@ build/proto.stamp: $(source_files)
 
-$(swagger_out): $(swagger_in) build/swagger
-	protoc -I src $< --swagger_out=logtostderr=true:build/swagger
+$(swagger_out): $(source_files) build/swagger
```
@rb-determined-ai should look at this not me.

hamidzr replied:
I this I should also update the comment on line 10

rb-determined-ai replied:
yes

stoksc commented on proto/scripts/swagger.py:
```diff
@@ -36,12 +36,6 @@ def clean(path: str, patch: str) -> None:
     with open(path, "r") as f:
         spec = json.load(f)
 
-    # Update path names to be consistent.
-    paths = {}
-    for key, value in spec["paths"].items():
-        paths[key.replace(".", "_")] = value
-    spec["paths"] = paths
-
```
so it was this + collisions?

hamidzr replied:
this added the collisions where `patch /abc/model.name` and `get /abc/model_name` collieded

rb-determined-ai replied:
Why was this originally added, do you know?

rb-determined-ai replied:
@hamidzr I'm still trying to understand the implications of this change.  I see a code diff in the generated files like this:

```diff
<     "/api/v1/checkpoints/{checkpoint.uuid}/metadata": {
---
>     "/api/v1/checkpoints/{checkpoint_uuid}/metadata": {
```

But I don't understand the significance.  Is that just purely cosmetic or does it affect the resulting API somehow?

hamidzr replied:
hmm my understanding is that in cases where it didn't used to cause collisions, this was purely visual. I can't say what each language plugin for swagger could do differently based on this change but the only binding we are actively using is working [fine](https://github.com/determined-ai/determined/pull/1437/checks?check_run_id=1249986030)

in general, I think if we don't have a good reason to modify this generated swagger spec, which is what this used to do,  we should avoid it as further demonstrated by the bug [here](https://determinedai.atlassian.net/browse/DET-4213)

rb-determined-ai replied:
cool

rb-determined-ai commented on proto/src/determined/api/v1/template.proto:
```diff
@@ -51,7 +51,7 @@ message GetTemplateResponse {
 message PutTemplateRequest {
   // The template to put.
-  determined.template.v1.Template template = 2;
+  determined.template.v1.Template template = 1;
```
Isn't this a breaking change?  Or... how does this work?

hamidzr replied:
this would be for anyone directly connecting to the grpc server which we don't have any AFAIK so I'm just cleaning it up here. we'll add the breaking change check here https://determinedai.atlassian.net/browse/DET-4209

hamidzr replied:
a sneak peak
```
hrz ➜ proto git:(master) ✗ buf check breaking --against-input /tmp/buf.image.bin                                            
src/determined/api/v1/template.proto:52:1:Previously present field "2" with name "template" on message "PutTemplateRequest" was deleted.

```

rb-determined-ai replied:
Non-blocking, but it seems like it doesn't buy us anything to make this change, while it is a breaking change.  I'd just as soon leave it alone on principle.  Non-blocking though because it's probably true that nobody is going to hit this.

hamidzr replied:
if we think we want to order fields sequentially and that it is common practice and might be more developer-friendly, now would be a good time to order them especially for messages where we have not intentionally reserved fields (https://developers.google.com/protocol-buffers/docs/proto3#assigning_field_numbers). In a lot of our messages, the field numbers are out of order just because they have been heavily under development and we haven't published gprc APIs yet and no other reason. with that said I'm fine with leaving it as is. @stoksc wdy think?

rb-determined-ai replied:
as I said, "non-blocking".

hamidzr commented:
> Why was this originally added, do you know?

from what I can tell so that the parameter names show up similar to each other in the UI.

hamidzr commented:
I rebased and removed a comment. this should now _fail_ the backward compatibility check.

seanr15 commented:
Will this be merged soon? This is needed for the SDK.

stoksc commented:
@seanr15 I think it's just waiting on the CI fix to land.

hamidzr commented:
> I rebased and removed a comment. this should now _fail_ the backward compatibility check.

https://app.circleci.com/pipelines/github/determined-ai/determined/6559/workflows/2c765ec2-5669-4d2b-89fb-c565d5fd2cdd/jobs/172831
